lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/Task/TaskRuntime.cs

```diff
@@ -220,6 +222,15 @@ public void Deliver(byte[] message)
         public void OnNext(ICloseEvent value)
         {
             Logger.Log(Level.Info, "TaskRuntime::OnNext(ICloseEvent value)");
+            try
+            {
+                _closeHandlerFuture.Get().OnNext(value);
+            }
+            catch (Exception ex)
+            {
+                var closeEx = new TaskCloseHandlerNotBoundException("Unable to close task.", ex);
+                Utilities.Diagnostics.Exceptions.CaughtAndThrow(closeEx, Level.Error, Logger);
```

Comment:
This exception shouldn't be related to `TaskCloseHandlerNotBoundException`: You bind a default handler, so there is no way to have a situation where a handler isn't bound. In this case, the handler was in fact bound, but threw an exception in its own right. This should result in an `IFailedTask` on the Driver side which contains enough information such that the Driver knows that its call to `IRunningTask.Dispose()` initiated this failure.